From 3c0eb6cd9e0438e0e9b6baafbb34e0446c25ab6f Mon Sep 17 00:00:00 2001
From: Richard Kuo <rkuo@codeaurora.org>
Date: Wed, 21 Oct 2015 13:57:53 -0500
Subject: [PATCH 10/22] Hexagon: enable dummy console if configured

Signed-off-by: Richard Kuo <rkuo@codeaurora.org>
---
 arch/hexagon/kernel/setup.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/arch/hexagon/kernel/setup.c b/arch/hexagon/kernel/setup.c
index fc552d1..6863490 100644
--- a/arch/hexagon/kernel/setup.c
+++ b/arch/hexagon/kernel/setup.c
@@ -100,6 +100,11 @@ void __init setup_arch(char **cmdline_p)
 #ifdef CONFIG_SMP
 	smp_start_cpus();
 #endif
+
+#if defined(CONFIG_DUMMY_CONSOLE)
+	conswitchp = &dummy_con;
+#endif
+
 }
 
 /*
-- 
1.8.4.3

